### DISCUSSION TITLE
Add button group styles for radio buttons

### DISCUSSION THREAD

**Discussion by ryanleichty** [craftcms/cms#15837] [Ideas] [answered]
We often use radio groups over drop down menus for the improved UX when you have few items to select from. It would be amazing if you could alternatively display the radio buttons as a button group with text only, icon only (using the font awesome library), or both.

![CleanShot 2024-10-03 at 12 22 10@2x](https://github.com/user-attachments/assets/46dac961-a4c7-44b5-999e-669ac6237a40)

![CleanShot 2024-10-03 at 12 05 52 2@2x](https://github.com/user-attachments/assets/4870f874-e857-4ca8-88de-f33c025824b2)

It's kind of similar to this https://github.com/craftcms/cms/discussions/15804 in that it's providing an alternative UI for the same kind of input.

I know some third party plugins exist for this, but it would be great if this were natively supported.

**Comment by daltonrooney** [comment]
I agree, this would be a nice improvement. I would also love an option to align the default radio field vertically or horizontally. Sometimes a horizontal radio list is a better use of space.

**Comment by ryanleichty** [comment]
Now that we have [icons for dropdown fields](https://github.com/craftcms/cms/pull/16645), do you think something like this would be possible for radio buttons @brandonkelly?

**Comment by brandonkelly** [comment, marked as answer]
Just PR’d this for Craft 5.7: #16782

**Reply by ryanleichty** [reply]
Awesome, you guys are cooking! 🔥

**Reply by alexanderbuergin** [reply]
@brandonkelly Would it be possible to convert Radio or Select Fields to this new field type?

**Reply by brandonkelly** [reply]
@alexanderbuergin Yep! Made even easier in 5.7 thanks to #16783.

**Reply by alexanderbuergin** [reply]
Thanks!

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with the new Button Group field type 🎉
